### PR TITLE
Add auto merge for pre commit and dependabot in a CI job

### PR DESCRIPTION
**Pull request by uittenbroekrobbert**
# Description

Add a CI job to auto merge updates done by dependabot and pre-commit. Merges for dependabot are only done if minor or patch. For pre-commit it updates everything, as we don't have that level of control.

Resolves #

https://github.com/orgs/MinBZK/projects/7?pane=issue&itemId=65105178

## Checklist

Please check all the boxes that apply to this pull request using "x":

Most changes are GitHub actions and those can not be tested locally. Pre-commit has been tested locally.

- [ ]  I have tested the changes locally and verified that they work as expected.
- [X]  I have followed the project's coding conventions and style guidelines.
- [X]  I have rebased my branch onto the latest commit of the main branch.
- [X]  I have squashed or reorganized my commits into logical units.
- [X]  I have read, understood and agree to the [Developer Certificate of Origin](../blob/main/DCO.md), which this project utilizes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1,9 @@
 * @MinBZK/ai-validation-team
+
+# excluded files so github actions bot can auto update dependencies
+#  see https://github.com/orgs/community/discussions/23064
+poetry.lock
```
Comment (berrydenhartog) [inline .github/CODEOWNERS:5]: Nice solution. Did not know. this one :D

File: .github/workflows/automerge.yml [hunk 2]
```diff
@@ -23,9 +22,30 @@ jobs:
...
+      - name: Enable Auto-Approve for Pre-Commit
+        if: github.head_ref == 'pre-commit-ci-update-config'
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{github.event.pull_request.html_url}}
+          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Enable Auto-Merge for Pre-Commit
+        if: github.head_ref == 'pre-commit-ci-update-config'
+        run: gh pr merge --auto --merge "$PR_URL"
```
Comment (berrydenhartog) [inline .github/workflows/automerge.yml:48]: Nice! only problem is that we require projects to have automerge enabled but is fine

**Review by berrydenhartog — COMMENTED** [review]
Could you also add a CI workflow that build/lint the project?

**Comment by uittenbroekrobbert** [comment]
> Could you also add a CI workflow that build/lint the project?

@berrydenhartog 

I assume this would be steps to perform before approve / merge, so it becomes something like:
steps:
  - checkout project
  - lint project (abort if failed)
  - build project (abort if failed)
  - (optional) test project (abort if failed)
  - approve
  - merge

Is it worth looking into reusable workflows (have you done that before?), it seems like a good way to avoid code duplication but is some more work to set up:
https://docs.github.com/en/actions/using-workflows/reusing-workflows

**Comment by berrydenhartog** [comment]
> > Could you also add a CI workflow that build/lint the project?
> 
> @berrydenhartog
> 
> I assume this would be steps to perform before approve / merge, so it becomes something like: steps:
> 
>     * checkout project
> 
>     * lint project (abort if failed)
> 
>     * build project (abort if failed)
> 
>     * (optional) test project (abort if failed)
> 
>     * approve
> 
>     * merge
> 
> 
> Is it worth looking into reusable workflows (have you done that before?), it seems like a good way to avoid code duplication but is some more work to set up: https://docs.github.com/en/actions/using-workflows/reusing-workflows

Indeed. but the approve and merge is done by your current action. 
in another action you could implement a very simple ci job that builds the package and runs a linter

**Comment by berrydenhartog** [comment]
Reusable workflows could be usefull, but often i do not see them used because everyone wants control over there pipeline.

**Review by berrydenhartog — APPROVED** [review]
LGTM

**Review by berrydenhartog — APPROVED** [review]
Lgtm

**Pull request merged**